ncrypt/crypt_gpgme.c

```diff
@@ -1404,6 +1407,39 @@ static void print_smime_keyinfo(const char *msg, gpgme_signature_t sig,
   }
 }
 
+/**
+ * show_encryption_info - Show encryption information
+ */
+static void show_encryption_info(struct State *state, gpgme_decrypt_result_t result)
+{
+    if (state->flags & STATE_DISPLAY)
+      state_attach_puts(state, _("[-- Begin encryption information --]\n"));
+
+    for (gpgme_recipient_t r = result->recipients; r; r = r->next)
+      show_one_recipient(state, r);
+
+    if (state->flags & STATE_DISPLAY)
+      state_attach_puts(state, _("[-- End encryption information --]\n\n"));
+}
+
+/**
+ * show_one_recipient - Show information about one encryption recipient
+ */
+static void show_one_recipient(struct State *state, gpgme_recipient_t r)
+{
+  const char  *algo;
+
+  algo = gpgme_pubkey_algo_name(r->pubkey_algo);
+  if (algo == NULL)
+    algo = "?";
+
+  state_puts(state, "Recipient: ");
```

Comment:
@flatcap 

When I press `v` from the pager (to view attachments), this line provokes a segmentation fault.

If you apply the following debugging diff on top of this patch, you'll see that the segfault is precisely there (the first one prints, but it crashes before the second one):

```
diff --git a/ncrypt/crypt_gpgme.c b/ncrypt/crypt_gpgme.c
index 247166179..d2b05dec7 100644
--- a/ncrypt/crypt_gpgme.c
+++ b/ncrypt/crypt_gpgme.c
@@ -1432,8 +1432,12 @@ static void show_one_recipient(struct State *state, gpgme_recipient_t r)
   algo = gpgme_pubkey_algo_name(r->pubkey_algo);
   if (algo == NULL)
     algo = "?";
+fprintf(stderr, "ALX  %s(): %d\n", __func__, __LINE__);
+getchar();
 
   state_puts(state, "Recipient: ");
+fprintf(stderr, "ALX  %s(): %d\n", __func__, __LINE__);
+getchar();
   state_puts(state, algo);
   state_puts(state, " key, ID ");
   state_puts(state, r->keyid);
@@ -1779,6 +1783,7 @@ restart:
   ciphertext = NULL;
 
   result = gpgme_op_decrypt_result(ctx);
+fprintf(stderr, "ALX  %s(): %d\n", __func__, __LINE__);
   if (result)
     show_encryption_info(state, result);
 
@@ -2583,6 +2588,7 @@ int pgp_gpgme_application_handler(struct Body *b, struct State *state)
           gpgme_decrypt_result_t result = NULL;
 
           result = gpgme_op_decrypt_result(ctx);
+fprintf(stderr, "ALX  %s(): %d\n", __func__, __LINE__);
           if (result)
             show_encryption_info(state, result);
 
```

(You'll also notice that the call is done from *decrypt_part*().)

I guess the screen is in a state that doesn't want to print this stuff, but do you know/suspect why?  What can we do about it?

Reply:
Maybe I should *state_attach_puts*() there too?  I don't know what each of those do.

Reply:
Or maybe check `(state->flags & STATE_DISPLAY)`?  Or maybe both?  I don't really understand those flags too much yet.

Reply:
The crash has been fixed.  I'll resolve this thread.